dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/ClassLoaders.scala

```diff
@@ -48,11 +48,29 @@ class DelegatingClassLoader(commonLoader: ClassLoader, sharedClasses: Set[String
       findResourcesMethod.invoke(cl, name).asInstanceOf[util.Enumeration[URL]]
     }
     val superResources = super.getResources(name)
-    val resources = new util.Vector[URL]()
-    while (appResources.hasMoreElements) resources.add(appResources.nextElement())
-    while (superResources.hasMoreElements) resources.add(superResources.nextElement())
-    resources.elements()
+    if (!superResources.hasMoreElements) {
+      appResources
+    } else if (!appResources.hasMoreElements) {
+      superResources
+    } else {
+      val resources = new util.HashSet[URL]
+      while (appResources.hasMoreElements) resources.add(appResources.nextElement())
+      while (superResources.hasMoreElements) resources.add(superResources.nextElement())
+      new util.Vector(resources).elements()
```

Comment:
Not sure how critical this is, but `Collections.enumeration(resources)` would avoid the extra copy.

Reply:
Good idea.

Reply:
Cool, want me to do it or are you working on this already?